tag.go

```diff
@@ -0,0 +1,150 @@
+package gomaasapi
+
+import (
+	"net/url"
+
+	"github.com/juju/errors"
+	"github.com/juju/schema"
+	"github.com/juju/version"
+)
+
+type tag struct {
+	controller  *controller
+	resourceURI string
+
+	name       string
+	definition string
+	comment    string
+}
+
+// Name implements Tag.
+func (s *tag) Name() string {
+	return s.name
+}
+
+// Definition implements Tag.
+func (s *tag) Definition() string {
+	return s.definition
+}
+
+// Comment implements Tag.
+func (s *tag) Comment() string {
+	return s.definition
+}
+
+func (s *tag) Machines() ([]Machine, error) {
```

Comment:
Another way of doing this is to keep the tag object as a pure domain object and have the api to get machines for a given tag as a method on the controller (or even as a method on the machine). A tag is really just a snippet of domain information. Same with AddToMachine() etc; they perhaps are better done as AddTag() and RemoveTag() methods on the machine entity.